### PR TITLE
[pkg/ottl] Add isList OTTL Function

### PR DESCRIPTION
**Pull request by strawgate**
Description:
Add isList OTTL function that returns true if the target is a slice and returns false if it is not

Link to tracking Issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27870

Testing: Added tests

Documentation: Updated readme and added changelog entry

### CODE REVIEW

File: pkg/ottl/ottlfuncs/func_is_list.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		switch valType := val.(type) {
+
+		case pcommon.Value:
+			switch valType.Type() {
+			case pcommon.ValueTypeSlice:
+				return true, nil
+			}
+			return false, nil
+
+		case pcommon.Slice, plog.LogRecordSlice, plog.ResourceLogsSlice, plog.ScopeLogsSlice, pmetric.ExemplarSlice, pmetric.ExponentialHistogramDataPointSlice, pmetric.HistogramDataPointSlice, pmetric.MetricSlice, pmetric.NumberDataPointSlice, pmetric.ResourceMetricsSlice, pmetric.ScopeMetricsSlice, pmetric.SummaryDataPointSlice, pmetric.SummaryDataPointValueAtQuantileSlice, ptrace.ResourceSpansSlice, ptrace.ScopeSpansSlice, ptrace.SpanEventSlice, ptrace.SpanLinkSlice, ptrace.SpanSlice:
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/func_is_list.go:52]: I think we should handle these types here as well:
```
[]string, []bool, []int64, []float64, [][]byte, []any
```

**Review by evan-bradley — COMMENTED** [review]
Thanks for taking this. Could you also put a test in `pkg/ottl/e2e/e2e_test.go` in the `Test_e2e_converters` function?

**Comment by strawgate** [comment]
Incorporated provided feedback!

File: pkg/ottl/ottlfuncs/func_is_list.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	if !ok {
+		return nil, fmt.Errorf("IsListFactory args must be of type *IsListArguments[K]")
+	}
+
+	return isList(args.Target), nil
+}
+
+func isList[K any](target ottl.Getter[K]) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (interface{}, error) {
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/func_is_list.go:37]:
```suggestion
	return func(ctx context.Context, tCtx K) (any, error) {
```
I _think_ we standardized on `any` in the repo, but can't recall for sure.

File: pkg/ottl/ottlfuncs/func_is_list.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		val, err := target.Get(ctx, tCtx)
+		if err != nil {
+			return false, err
+		}
+
+		switch valType := val.(type) {
+
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/func_is_list.go:44]:
```suggestion
		switch valType := val.(type) {
```
Nit, but I think typically there's not a newline here.

**Review by evan-bradley — APPROVED** [review]
Thanks!

**Comment by strawgate** [comment]
Incorporated feedback and updated branch

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**